src/lib/ledger_builder/ledger_builder.ml

```diff
@@ -1064,7 +1138,10 @@ end = struct
       Or_error.ok_exn (fill_in_completed_work t.scan_state works)
     in
     Or_error.ok_exn (Parallel_scan.enqueue_data ~state:t.scan_state ~data) ;
-    Or_error.ok_exn (verify_scan_state_after_apply t.ledger t.scan_state) ;
+    let%map () =
```

Comment:
Verify_scan_state just checks if the ordering is right (basically the initial version of scan_statement we had). I added it here just as an extra check to make sure the diff when applied doesn't mess with the ordering in the tree. But we shouldn't be doing the proof verification part because we already do that before adding it to the diff

Reply:
@johnwu93 does this make sense?